exercise-integration-testing-5-53/test/spread-the-code-for-task-10/when-event-is-valid.test.js

```diff
@@ -0,0 +1,270 @@
+    });
+  
+    // ✅ TASK: Test that when a new valid event is posted to /sensor-events route, it's indeed retrievable from the DB
+    // 💡 TIP: Whenever possible, use the public API for verification
+  
+    test("When new valid event is send,  should retrievable the data from the DB (2 objects)", async () => {
+      //Arrange
+      const eventToAdd = {
+        category: "kids-room",
+        temperature: 20,
+        name: "Thermostat-temperature-new", //This must be unique
+        color: "Green",
+        weight: "80 gram",
+        status: "active",
+      };
+      //Act
+  
+      const config = {
+        validateStatus: () => true,
+      };
+      const urlFetchSensorEvent = `http://localhost:${port}/sensor-events`;
+      const urlGetEventByCategory = `http://localhost:${port}/sensor-events/${eventToAdd.category}/${eventToAdd.name}`;
+  
+      await axios.post(urlFetchSensorEvent, eventToAdd, config);
+      const eventFromDb = await axios.get(urlGetEventByCategory);
+      // console.log("retrive data ", eventFromDb);
+      //Assert
+      expect(eventFromDb.status).toBe(200);
+      expect(eventFromDb.data.length).toBe(2); // One obj from the last test and one is from the current test.
+    });
+  
+    // ✅ TASK: Test that querying the GET:/sensor-events route, it returns the right event when a single event exist
+    // 💡 TIP: Ensure that exactly one was returned and that this is the right event
+    // 💡 TIP: Try using as few assertions as possible, maybe even only one
+  
+    test("When new valid event is send, it return the right event when a single event exist", async () => {
+      //Arrange
+  
+      //Act
+  
+      const eventToAdd = {
+        category: `${helper(5)}`,
+        temperature: 20,
+        name: "Thermostat-temperature-new", //This must be unique
+        color: "Green",
+        weight: "80 gram",
+        status: "active",
+      };
+  
+      console.log(eventToAdd);
+      const config = {
+        validateStatus: () => true,
+      };
+      const urlFetchSensorEvent = `http://localhost:${port}/sensor-events`;
+      const urlGetEventByCategory = `http://localhost:${port}/sensor-events/${eventToAdd.category}/${eventToAdd.name}`;
+  
+      await axios.post(urlFetchSensorEvent, eventToAdd, config);
+      const eventFromDb = await axios.get(urlGetEventByCategory);
+      console.log("retrive data ", eventFromDb.data.length);
+      //Assert
+      expect(eventFromDb.status).toBe(200);
+      expect(eventFromDb.data.length).toBe(1); // One obj from the last test and one is from the current test.
+    });
+  
+    // ✅ TASK: Test that querying the GET:/sensor-events route, it returns the right events when multiple events exist
+    // 💡 TIP: Ensure that all the relevant events were returned
+  
+    test("When new valid event is send, should retrievable the data from the DB when muliple events exist", async () => {
+      //Arrange
+  
+      const category = helper(5);
```

Comment:
Important principle: Let the reader understand without leaving the test and navigating to another code. The name 'helper' does not tell me much about what is going on there, I'm being tempted to navigate and read it